src/component.vue

```diff
@@ -90,19 +83,25 @@ export default {
     }
   },
   watch: {
-    value(newValue) {
-      // avoid trigering the directive's update hook when we emit
-      // the unmasked value to the parent component
-      if (newValue !== this.emittedValue) {
-        this.maskedValue = newValue
-      }
+    modelValue: {
+      handler(newValue) {
+        // avoid trigering the directive's update hook when we emit
+        // the unmasked value to the parent component
+        if (newValue !== this.emittedValue) {
+          this.maskedValue = newValue
+        }
+      },
+      deep: true
```

Comment:
(Optional) Prefer if we keep `deep: true` above `handler` so it's easy to see at a glance what flags are turned on for the watchers when starting to read from the top. Same for elsewhere.